appscale/tools/agents/ec2_agent.py

```diff
@@ -191,43 +213,70 @@ def create_security_group(self, parameters, group):
       "name {0}".format(group))
 
 
-  def authorize_security_group(self, parameters, group, from_port, to_port,
-    ip_protocol, cidr_ip):
+  def get_security_group_by_name(self, conn, group):
+    """Gets a security group in AWS with the given name.
+
+    Args:
+      conn: A boto connection.
+      group: A str that names the group that should be found.
+    Returns:
+      The 'boto.ec2.securitygroup.SecurityGroup' that has the correct group
+      name.
+    Raises:
+      SecurityGroupNotFoundException: If the security group could not be found.
+    """
+    try:
+      return next(sg for sg in conn.get_all_security_groups()
+                  if sg.name == group)
+    except StopIteration:
+      raise SecurityGroupNotFoundException(
+          'Could not find security group with name {}!'.format(group))
+
+
+  def authorize_security_group(self, parameters, group_id, from_port,
+                               to_port, ip_protocol, cidr_ip, group_name=None):
     """Opens up traffic on the given port range for traffic of the named type.
 
     Args:
       parameters: A dict that contains the credentials necessary to authenticate
         with AWS.
-      group: A str that names the group whose ports should be opened.
+      group_id: A str that contains the id of the group whose ports should be
+        opened.
       from_port: An int that names the first port that access should be allowed
         on.
       to_port: An int that names the last port that access should be allowed on.
       ip_protocol: A str that indicates if TCP, UDP, or ICMP traffic should be
         allowed.
       cidr_ip: A str that names the IP range that traffic should be allowed
         from.
+      group_name: A str that contains the name of the group whose ports
+        should be opened. Default is None since EC2 can just use group_id but
+        Euca will override to the group name in the parameters. EC2 cannot
+        use both group_id and group_name.
```

Comment:
It seems simpler to use group_id in all cases.

Reply:
group_id didn't appear to work for me in euca for classic networking, so ec2 is using group_id always but euca calls the method using the group name.

Reply:
I think the issue is that the euca agent is configured to use the `2010-08-31` ec2 api which was different from the one being used in the current `boto` client. You could probably just bump the euca agent to use the current api version '2016-11-15'.